src/Microsoft.ML.Data/EntryPoints/PredictorModelImpl.cs

```diff
@@ -125,7 +125,7 @@ internal override string[] GetLabelInfo(IHostEnvironment env, out ColumnType lab
             if (trainRms.Label != null)
             {
                 labelType = trainRms.Label.Value.Type;
-                if (labelType is KeyType keyType && trainRms.Label.Value.HasKeyValues(keyType.Count))
```

Comment:
>labelType is KeyType keyType [](start = 20, length = 28)

Simplifying this may be acceptable if we also simultaneously fix the bug in `HasKeyValues` that I see we have (at least in `master` right now) that `HasKeyValues` will succeed even if the input column is not a key type. :)

Reply:
It will return false in current version because GetKeyCount() returns 0 when type is not KeyType.

---
In reply to: [249213625](https://github.com/dotnet/machinelearning/pull/2146#discussion_r249213625) [](ancestors = 249213625)

Reply:
Ah that makes sense. Thanks @artidoro .